plugins/ui/edgy.nix

```diff
@@ -0,0 +1,321 @@
+        '';
+
+        on_begin = helpers.defaultNullOpts.mkLuaFn ''
+          function()
+            vim.g.minianimate_disable = true
+          end
+        '' "Callback for the beginning of animations.";
+
+        on_end = helpers.defaultNullOpts.mkLuaFn ''
+          function()
+            vim.g.minianimate_disable = false
+          end
+        '' "Callback for the ending of animations.";
+
+        spinner = mkOption {
+          type =
+            with helpers.nixvimTypes;
+            nullOr (
+              either strLua (submodule {
+                freeformType = attrsOf anything;
+                options = {
+                  frames = helpers.defaultNullOpts.mkListOf types.str [
+                    "⠋"
+                    "⠙"
+                    "⠹"
+                    "⠸"
+                    "⠼"
+                    "⠴"
+                    "⠦"
+                    "⠧"
+                    "⠇"
+                    "⠏"
+                  ] "Frame characters.";
+
+                  interval = helpers.defaultNullOpts.mkUnsignedInt 80 ''
+                    Interval time between two consecutive frames.
+                  '';
+                };
+              })
+            );
+          default = null;
+          example = "require('noice.util.spinners').spinners.circleFull";
+          apply = v: if isString v then helpers.mkRaw v else v;
+          description = ''
+            Spinner for pinned views that are loading.
+
+            ${helpers.defaultNullOpts.mkDefaultDesc {
+              frames = [
+                "⠋"
+                "⠙"
+                "⠹"
+                "⠸"
+                "⠼"
+                "⠴"
+                "⠦"
+                "⠧"
+                "⠇"
+                "⠏"
+              ];
+              interval = 80;
+            }}
+          '';
+        };
+      };
+
+      exit_when_last = helpers.defaultNullOpts.mkBool false ''
+        Enable this to exit Neovim when only edgy windows are left.
+      '';
+
+      close_when_all_hidden = helpers.defaultNullOpts.mkBool true ''
+        Close edgy when all windows are hidden instead of opening one of them.
+        Disable to always keep at least one edgy split visible in each open section.
+      '';
+
+      wo = helpers.defaultNullOpts.mkAttrsOf types.anything {
+        winbar = true;
+        winfixwidth = true;
+        winfixheight = false;
+        winhighlight = "WinBar:EdgyWinBar,Normal:EdgyNormal";
+        spell = false;
+        signcolumn = "no";
+      } "Global window options for edgebar windows.";
+
+      keys = mkOption {
+        type = with helpers.nixvimTypes; attrsOf (either strLuaFn (enum [ false ]));
+        default = { };
+        apply = mapAttrs (_: v: if isString v then helpers.mkRaw v else v);
+        description = ''
+          Buffer-local keymaps to be added to edgebar buffers.
+          Existing buffer-local keymaps will never be overridden.
+          Set to `false` to disable a builtin.
+
```

Comment:
Not sure how we usually do this, but it'd be nice to document the expected function signature, i.e. a void function that accepts one parameter "`win`" which is a `Edgy.Window`

Reply:
Like so ?